src/main/java/Main.java

```diff
@@ -1,22 +1,30 @@
+import java.io.BufferedWriter;
 import java.io.File;
 import java.io.FileWriter;
 import java.io.IOException;
 
 public class Main {
 
     public static void main(String[] args) throws IOException {
-
+        Company company = new Company();
         File file = new File("employees.csv");
 
         if (file.exists()) {
             File stats = new File("stats.txt");
             stats.createNewFile();
+            company.readEmployees();
```

Comment:
To jest trochę dziwne, raczej unikamy tworzenia kodu, gdzie trzeba zachować odpowiednią sekwencję przy jego uruchamianiu. Tutaj tą sekwencją jest utworzenie obiektu, a potem wywołanie na niej `readEmployees()`. Fajniejsza byłaby taka konstrukcja

`Company company = CompanyReader.readEmployees();`, gdzie wszystko dzieje się w jednym etapie z punktu widzenia użytkownika

Reply:
w konstrukcji, którą przedstawiłeś nie rozumiem w jaki sposób miałoby to powstać, ponieważ Company company InteliJ krzyczał, że oczekuje obiektu Company, a ja próbuję przekazać mu referencję do tablicy. 
Zmieniłem ten fragment kodu tak, żeby bez tworzenia obiektu company przypisać tablicę z metody readEmployees i przekazałem ją do następnych metod, które także oznaczyłem jako statyczne, żeby uniknąć tworzenia obiektu company.

Reply:
No właśnie średnio to teraz wygląda. Dostajesz `Employee[]` i muisisz ją teraz przekazywać kolejno do każdej metody. Lepiej byłoby gdyby `Company` miało w sobie pole z `Employee[]`, przez co każda metoda tej klasy miałaby dostęp do pola.